vnext/Microsoft.ReactNative/ReactPropertyBag.idl

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.ReactNative {
+
+  // The delegate is used to create property value on-demand.
+  [webhosthidden]
+  delegate Object ReactCreatePropertyValue();
+
+  // Namespace for the property name.
+  // Use ReactPropertyNameTable.GetNamespace to get atomic property namespace for a string.
+  [webhosthidden]
+  interface IReactPropertyNamespace {
+    // Get String name representation of the property namespace.
+    String NamespaceName { get; };
+  }
+
+  // Name of a property.
+  // Use ReactPropertyNameTable.GetName to get atomic property name for a string.
+  [webhosthidden]
+  interface IReactPropertyName {
+    // The local property name String in context of the property namespace.
+    String LocalName { get; };
+
+    // The namespace the property name is defined in.
+    IReactPropertyNamespace Namespace { get; };
+  }
+
+  // The IPropertyBag provides a thread-safe property storage.
+  // Properties are identified by a propertyName.
+  // have their own set of properties not visible to other components.
+  // It is expected that there will be no direct use of this interface.
+  // All usage should happen through a strongly typed accessors.
+  [webhosthidden]
+  runtimeclass ReactPropertyBag {
+    // Construct new PropertyNameTable.
```

Comment:
>PropertyNameTable [](start = 21, length = 17)

Nit: Old name?

Reply:
Yep... I went through several iterations. I will fix it. Thank you!